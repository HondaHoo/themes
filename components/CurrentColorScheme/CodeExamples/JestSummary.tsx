import {
  Foreground,
  Background,
  Red,
  Green,
  Yellow,
} from '@/components/CurrentColorScheme/CodeExamples/ColorComponents';

const Gaps = ({count}: {count: number}) => <>{Array(count).fill(' ')}</>;

const JestSummary = () => {
  return (
    <figure>
      <figcaption>Jest summary example</figcaption>
      <Background>
        <div>
          <Green bold inverse>
            {' '}
            PASS{' '}
          </Green>{' '}
          <Foreground dim>components/ToggleCurrentLightness/</Foreground>
          <Foreground bold>ToggleCurrentLightness.test.tsx</Foreground>
        </div>
        <div>
          <Red bold inverse>
            {' '}
            FAIL{' '}
          </Red>{' '}
          <Foreground dim>components/NextPrevButton/</Foreground>
          <Foreground bold>NextPrevButton.test.tsx</Foreground>
        </div>
        <div>
          <Foreground bold>Test Suites: </Foreground>
          <Red bold>1 failed</Red>, <Green bold>9 passed</Green>, 10 total
        </div>
        <div>
          <Foreground bold>
            Time: <Gaps count={6} />
          </Foreground>{' '}
          <Yellow bold>2.897</Yellow>
        </div>
        <div>
          <Foreground dim>Ran all test suites.</Foreground>
        </div>
      </Background>
    </figure>
  );
};

export default JestSummary;
