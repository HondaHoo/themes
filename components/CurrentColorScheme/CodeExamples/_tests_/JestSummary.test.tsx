import '@testing-library/jest-dom';
import {render, screen} from '@testing-library/react';

import JestSummary from '../JestSummary';

it('should render all the components', () => {
  render(<JestSummary />);
  expect(screen.getByRole('figure')).toMatchSnapshot();
});
