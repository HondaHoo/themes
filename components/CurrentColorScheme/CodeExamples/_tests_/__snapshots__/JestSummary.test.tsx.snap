// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render all the components 1`] = `
<figure>
  <figcaption>
    Jest summary example
  </figcaption>
  <div
    class="background"
  >
    <div>
      <span
        class="inverse bold"
        style="--color: var(--green);"
      >
         
        PASS
         
      </span>
       
      <span
        class="foreground dim"
      >
        components/ToggleCurrentLightness/
      </span>
      <span
        class="foreground bold"
      >
        ToggleCurrentLightness.test.tsx
      </span>
    </div>
    <div>
      <span
        class="inverse bold"
        style="--color: var(--red);"
      >
         
        FAIL
         
      </span>
       
      <span
        class="foreground dim"
      >
        components/NextPrevButton/
      </span>
      <span
        class="foreground bold"
      >
        NextPrevButton.test.tsx
      </span>
    </div>
    <div>
      <span
        class="foreground bold"
      >
        Test Suites: 
      </span>
      <span
        class="red bold"
      >
        1 failed
      </span>
      , 
      <span
        class="green bold"
      >
        9 passed
      </span>
      , 10 total
    </div>
    <div>
      <span
        class="foreground bold"
      >
        Time: 
         
         
         
         
         
         
      </span>
       
      <span
        class="yellow bold"
      >
        2.897
      </span>
    </div>
    <div>
      <span
        class="foreground dim"
      >
        Ran all test suites.
      </span>
    </div>
  </div>
</figure>
`;
